tests/test_workers/test_tasks/test_build.py

```diff
@@ -445,10 +494,37 @@ def test_handle_add_request(
 
 @mock.patch('iib.workers.tasks.build.gate_bundles')
 @mock.patch('iib.workers.tasks.build._verify_labels')
-def test_handle_add_request_gating_failure(mock_vl, mock_gb):
+@mock.patch('iib.workers.tasks.build._get_resolved_bundles')
+def test_handle_add_request_gating_failure(mock_grb, mock_vl, mock_gb):
     error_msg = 'Gating failure!'
     mock_gb.side_effect = IIBError(error_msg)
+    mock_grb.return_value = ['some-bundle@sha']
+    bundles = ['some-bundle:2.3-1']
+    cnr_token = 'token'
+    organization = 'org'
+    greenwave_config = {'some_key': 'other_value'}
+    with pytest.raises(IIBError, match=error_msg):
+        build.handle_add_request(
+            bundles,
+            'binary-image:latest',
+            3,
+            'from-index:latest',
+            ['s390x'],
+            cnr_token,
+            organization,
+            False,
+            None,
+            greenwave_config,
+        )
+    mock_vl.assert_called_once()
+    mock_gb.assert_called_once_with(['some-bundle@sha'], greenwave_config)
 
+
+@mock.patch('iib.workers.tasks.build._verify_labels')
+@mock.patch('iib.workers.tasks.build._get_resolved_bundles')
+def test_handle_add_request_bundle_resolution_failure(mock_grb, mock_vl):
```

Comment:
I don't see any value in this test. I think it could be removed but I'd like opinions from others review this PR.

Reply:
I think there's no harm in being extra cautious but if everyone feels it can be removed, I will be happy to remove it. :)